client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/operations/PrepareTransactionOperation.java

```diff
@@ -80,8 +82,18 @@ protected void executeOperation(Channel channel) {
       channel.writeAndFlush(buf);
    }
 
+   @Override
+   protected void fetchChannelAndInvoke(int retryCount, Set<SocketAddress> failedServers) {
+      if (retryCount < modifications.size()) {
```

Comment:
I'd simplify this to only use the key owner on the first try.

IMO retries should be rare enough that we shouldn't worry about the server we're retrying on being a key owner, and we're going to send the request to an unrelated server anyway if the 2nd modification's key is owned by the failed server.

Reply:
I don't see any advantage in doing that. 
If there is a chance of hitting a primary owner of some key, I'll take it. 
If you are worried about `modifications.size()`, the list is an `ArrayList`; it just reads a field.

Reply:
I wasn't worried about `size()`, I was just thinking it's simpler if we use the same logic here as we use for other multi-key operations: 

https://github.com/infinispan/infinispan/blob/a2a183a3e23b37eb9820c7c18bec211ef2dc2deb/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/PutAllOperation.java#L76

Reply:
@danberindei  done.